cli/azd/internal/appdetect/dotnet_apphost.go

```diff
@@ -21,6 +21,29 @@ func (ad *dotNetAppHostDetector) Language() Language {
 }
 
 func (ad *dotNetAppHostDetector) DetectProject(ctx context.Context, path string, entries []fs.DirEntry) (*Project, error) {
+	// First, check for single-file apphost (apphost.cs)
+	for _, entry := range entries {
+		if entry.IsDir() {
+			continue
+		}
+
+		name := entry.Name()
+		// Check if this is a single-file apphost
+		if filepath.Ext(name) == ".cs" {
```

Comment:
Nit - is it worth deferring this check to `IsSingleFileAspireHost` so we only have one spot to update when we want to enable polyglot support?